pycaret/internal/preprocess/preprocessor.py

```diff
@@ -850,7 +850,7 @@ def _balance(self, fix_imbalance_method):
                     "Invalid value for the strategy parameter, got "
                     f"{fix_imbalance_method}. Choose from: {', '.join(strategies)}."
                 )
-            balance_estimator = FixImbalancer(strategies[fix_imbalance_method]())
+            balance_estimator = FixImbalancer(strategies[fix_imbalance_method](random_state=session_id))
```

Comment:
Not all estimators have a random_state parameter. Please check the estimator's signature before adding it.

Reply:
On it